# 03_qiime_imports Documentation
This is the third file to run in this distribution. It will take the assembled `.fna` file from 02_compile_sequences.sh and import this file, as well as a specified taxonomy and reference sequence file, into the QIIME 2 Bash environment. In this environment, QIIME will construct artifacts, which contain the `.qza` extension, that can then be ported into 04_Classify_Taxonomy.py for analysis.


Before running make sure to run `~ECEV32000_Project % chmod -x 03_qiime_imports.sh`
To run
```
~ECEV32000_Project % bash 03_qiime_imports.sh
```

Although the program accounts for this, one could also load the QIIME2 environment in their shell at this point by running
```
~% source activate qiime2-2020.2
```
This is assuming that QIIME2 has been installed already as per [the instructions from the developers](https://docs.qiime2.org/2020.2/install/native/)

```
#!/bin/bash
# Import sequences and taxonomy_ref into qiime format (Aritifacts)

fo="$(date +'%y_%m_%d')" # Assumes that downloading data using other scripts in this project
folder=$fo"_16s_Sequencing"

cd "data/$folder" # Specify directory of sequencing data

echo "Initialize qiime2"
```
### Change source of subshell to access qiime2
This will only change the subshell source to work in the QIIME2 environment.
```
# Change source to access qiime
source /Users/arnoldj/opt/anaconda3/bin/activate qiime2-2020.2
```
QIIME2 commands to convert the compiled sequences file to a QIIME2 artfact.
```
qiime tools import \
	--input-path sequences.fna \
	--output-path sequences.qza \
	--type 'FeatureData[Sequence]'
```
Importing the taxonomy data, downloaded from the [qiime2 greengenes database](http://qiime.org/home_static/dataFiles.html). As of this distribution, only the 61, 64, 67, 70, 73, 76, 79, 82, 85, 88 percent identity databases are available to upload to this github repository (due to filesize). If the greengenes database is downloaded directly from the provided website and added to the data folder created in step 01, the 91, 95, 97 and 99 percent identity databases will be accessible - for a future release.
```
# To change % Identity below, specify xx_otu_taxonomy.txt below
# Options include: 61, 64, 67, 70, 73, 76, 79, 82, 85, 88 working on uploading 91 and 94... darn filesize limits...
# Here we choose 85 % ID (from greengenes database)

qiime tools import \
	--input-format HeaderlessTSVTaxonomyFormat \
	--input-path 85_otu_taxonomy.txt \
	--output-path taxonomy_ref.qza \
	--type 'FeatureData[Taxonomy]'

qiime tools import \
  --type 'FeatureData[Sequence]' \
  --input-path 85_otus.fasta \
  --output-path 85_otus.qza
```
For the classifier created in step 04, reads must be extracted from the otus artifact.
```
qiime feature-classifier extract-reads \
  --i-sequences 85_otus.qza \
  --p-f-primer GTGCCAGCMGCCGCGGTAA \
  --p-r-primer GGACTACHVGGGTWTCTAAT \
  --p-trunc-len 120 \
  --o-reads ref-seqs.qza
```
